### PR TITLE
Bug 1846805: Pin all jsonnet dependencies to master

### PR DESCRIPTION
**Pull request by lilic**
Bump prometheus-operator to the v0.39.0 release.

* [x] I added CHANGELOG entry for this change.
* [ ] No user facing changes, so no entry in CHANGELOG was needed.

### CODE REVIEW

File: assets/node-exporter/daemonset.yaml [hunk 1]
```diff
@@ -25,8 +24,7 @@ spec:
         - --path.rootfs=/host/root
         - --no-collector.wifi
         - --no-collector.hwmon
-        - --collector.filesystem.ignored-mount-points=^/(dev|proc|sys|var/lib/docker/.+)($|/)
-        - --collector.filesystem.ignored-fs-types=^(autofs|binfmt_misc|cgroup|configfs|debugfs|devpts|devtmpfs|fusectl|hugetlbfs|mqueue|overlay|proc|procfs|pstore|rpc_pipefs|securityfs|sysfs|tracefs)$
+        - --collector.filesystem.ignored-mount-points=^/(dev|proc|sys|var/lib/docker/.+|var/lib/kubelet/pods/.+)($|/)
```
Comment (lilic) [inline assets/node-exporter/daemonset.yaml:27]: hmm should look up these changes if they have any effect on us.

**Comment by paulfantom** [comment]
I run `make generate-in-docker` locally and it produces different output than one from PR, but it matches one from CI.

**Comment by lilic** [comment]
> it matches one from CI.

it? Which one matches the CI yours or mine? Because generate said 🍏 ?

**Comment by paulfantom** [comment]
Strange, must have been something on my end. Sorry for the noise :)

/unhold

**Comment by lilic** [comment]
no worries @paulfantom PTAL :)

File: assets/node-exporter/daemonset.yaml (hunk 1 above)
Comment (paulfantom) [reply assets/node-exporter/daemonset.yaml:27]: This should be fine.

File: assets/node-exporter/service-monitor.yaml [hunk 2]
```diff
@@ -27,4 +27,3 @@ spec:
   selector:
     matchLabels:
       app.kubernetes.io/name: node-exporter
-      app.kubernetes.io/version: v0.18.1
```
Comment (paulfantom) [inline assets/node-exporter/service-monitor.yaml:30]: I am curious, why are we removing this downstream?

Comment (lilic) [reply assets/node-exporter/service-monitor.yaml:30]: This was an upstream change, not a downstream one.

Comment (paulfantom) [reply assets/node-exporter/service-monitor.yaml:30]: Yes, that's why I am asking why we are removing it downstream. I am aware why it was removed upstream (labels are immutable).

File: assets/prometheus-operator/cluster-role.yaml [hunk 3]
```diff
@@ -4,29 +4,9 @@ metadata:
...
-- apiGroups:
-  - apiextensions.k8s.io
-  resources:
-  - customresourcedefinitions
-  verbs:
-  - create
-- apiGroups:
-  - apiextensions.k8s.io
-  resourceNames:
-  - alertmanagers.monitoring.coreos.com
```
Comment (paulfantom) [inline assets/prometheus-operator/cluster-role.yaml:19]: If we remove this, what will be managing CRDs?

Comment (lilic) [reply assets/prometheus-operator/cluster-role.yaml:19]: Ah yes thanks! That explains your failure from your PR to bump prom-operator. Forgot it was this release that removes it. https://issues.redhat.com/browse/MON-1084 We have this in the next sprint, will just do it then, nothing urgent from this PR as mentioned in description.

/hold

**Comment by paulfantom** [comment]
/retitle Bug 1846805: Pin all jsonnet dependencies to master

**Comment by paulfantom** [comment]
/title Bug 1846805: Pin all jsonnet dependencies to master

**Comment by lilic** [comment]
Sadly this is still blocked until we bump prometheus operator again.

**Comment by paulfantom** [comment]
After syncing with `master` we should merge https://github.com/openshift/console/pull/5929

**Comment by simonpasquier** [comment]
/close

@lilic I suppose we don't need this PR anymore since we're aiming for v0.40.0 (hopefully!).

**Comment by lilic** [comment]
We still need to bump all the things to latest and then pin it, but yeah I think to not double the work you can do that in your PR as release Shepard?

**Pull request closed without merging**